docs/fundamentals/tutorials/new_executor.md

```diff
@@ -0,0 +1,370 @@
+
+from jina import Executor, DocumentArray, requests
+from jina.logging.logger import JinaLogger
+
+
+class RequestLogger(Executor):                                                                      # needs to inherit from Executor
+    def __init__(self,
+                default_log_docs: int = 1,                                                          # your arguments
+                *args, **kwargs):                                                                   # *args and **kwargs are required for Executor
+        super().__init__(*args, **kwargs)                                                           # before any custom logic
+        self.default_log_docs = default_log_docs
+        self.logger = JinaLogger('req_logger')
+        self.log_path = os.path.join(self.workspace, 'log.txt')
+        if not os.path.exists(self.log_path):
+            with open(self.log_path, 'w'): pass
+
+    @requests                                                                                       # decorate, by default it will be called on every request
+    def log(self,                                                                                   # arguments are automatically received
+            docs: Optional[DocumentArray],
+            parameters: Dict,
+            **kwargs):
+        self.logger.info('Request being processed...')
+
+        nr_docs = int(parameters.get('log_docs', self.default_log_docs))                            # accesing parameters (nr are passed as float due to Protobuf)
+        with open(self.log_path, 'a') as f:
+            f.write(f'request at time {time.time()} with {len(docs)} documents:\n')
+            for i, doc in enumerate(docs):
+                f.write(f'\tsearching with doc.id {doc.id}. content = {doc.content}\n')
+                if i + 1 == nr_docs:
+                    break
+```
+
+And that's it. We have an `Executor` that takes whatever Documents we pass to it and logs them. 
+
+Ok, and what now? How can you use this in your app?
+
+### Push your Executor to Hub
+
+We could use our Executor directly in our app, but here we will see how to push it to Jina Hub so we can share it with more people, or use it later. 
+
+First step is to actually make sure the `manifest.yml` and `config.yml` files are still relevant. Check that the data in there still represent you Executor's purpose.
+
+For this, you need to open a terminal in the folder of your `executor.py`, so in this case, open a terminal inside the `RequestLogger` folder. And there you just need to type:
+
+```bash
+jina hub push --public .
+```
+
+This means you will push your Executor publicly to Jina Hub. The last dot means you will use your current path. Once you run that command, you should see something like this:
+
+```{figure} ../../../.github/images/push-executor.png
+:align: center
+```
+
+Since we pushed our Executor using the `--public` flag, the only thing we will use is the ID, which, in this case, is `zsor7fe6`. 
```

Comment:
````suggestion
```{admonition} Note
:class: note
Since we pushed our Executor using the `--public` flag, the only thing we will use is the ID, which, in this case, is `zsor7fe6`. 
```
````
Maybe also explain what does public mean and reference the push to hub section